Dockerfile

```diff
@@ -0,0 +1,20 @@
+# build gradle project in first step
+FROM openjdk:8 AS BUILD_IMAGE
+LABEL maintainer = "Tobias Mathony <mathony.tobias@gmail.com>"
+COPY . .
+RUN ./gradlew build -x test --no-daemon
+
+FROM ubuntu:18.04
+RUN apt-get -qq update && apt-get install -qqy software-properties-common openjdk-8-jdk wget
```

Comment:
why don't you use a java image directly?

Reply:
I removed the first (unnecessary) build stage and moved the build of the gradle project instead to the ubuntu image to bundle it with the installation and linkage of prolog.

Reply:
Okay, I just thought of the openjdk-8 container as a base, but no worries :)